design/security-model.md

```diff
@@ -0,0 +1,360 @@
+# Security Model
+
+The Rook operator currently uses a highly privileged service account with permissions to create namespaces, roles, role bindings, etc. Our approach would not pass a security audit and this design explores an improvement to this. Furthermore given our use of multiple service accounts and namespace, setting policies and quotas is harder than it needs to be.
+
+## Goals
+ * Reduce the number of service accounts and privileges used by Rook
+ * Reduce the number of namespaces that are used by Rook
+ * Only use services accounts and namespaces used by the cluster admin -- this enables them to set security policies and quotas that rook adheres to
+ * Continue to support a least privileged model
+
+## What we do today
+
+Today the cluster admin creates the rook system namespace, rook-operator service account and RBAC rules as follows:
+
+```
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: rook-system
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: rook-operator
+  namespace: rook-system
+---
+kind: ClusterRole
+apiVersion: rbac.authorization.k8s.io/v1beta1
```

Comment:
it is v1 now 
https://kubernetes.io/docs/admin/authorization/rbac/

Reply:
will change this in the actual code.